Automattic/jetpack
# Likes: reimplement non-admin portions without jQuery

This PR reimplements the non-admin portions of Likes (namely `queuehandler.js`) without jQuery.

It also removes what appears to be deprecated functionality, in the form of a `slim-likes-widget` option. There appears to be no reference to this anywhere else in Jetpack or wpcom, which makes me think this may have been an early version of a different "slim" option that does exist today (`wpl-slim-likebox`) but isn't directly referenced in Jetpack.

`post-count.js` and `post-count-jetpack.js` are not touched in this PR, as they're only loaded in the context of `wp-admin`, which is not as much of a performance concern and thus out of scope for this PR.

#### Changes proposed in this Pull Request:

* Reimplement `queuehandler.js` without jQuery
* Remove apparently deprecated `slim-likes-widget` option
* Remove `jquery.jetpack-resize.js`, as the apparently deprecated `slim-likes-widget` option was the only thing that made use of it
* Change Jetpack plugin module linting to be ES2019-friendly
* Remove `queuehandler.js` and deleted `jquery.jetpack-resize.js` from ESLint exclusion list

#### Jetpack product discussion
None.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Ensure that the Likes and Comment Likes functionality continues to work correctly in Jetpack sites.
Please try different sites, as some of them use different code-paths; for example, some sites have a link that presents a popup with "other gravatars".
- Ensure that the Likes and Comment Likes settings pages continue to work correctly. Nothing is expected to have changed for these.
